src/Targets/Microsoft.Managed.DesignTime.targets

```diff
@@ -283,4 +278,15 @@
   <!-- This target collects all the resolved references that are used to actually compile. -->
   <Target Name="CollectResolvedCompilationReferencesDesignTime" DependsOnTargets="CompileDesignTime" Returns="@(ReferencePathWithRefAssemblies)" />
 
+  <!-- This target collects all up-to-date file additions. -->
+  <Target Name="CollectUpToDateInputsDesignTime" DependsOnTargets="CompileDesignTime" Returns="@(UpToDateCheckInput)">
+    <!-- If the project we're compiling has a ruleset, make sure we check it for up-to-date checks. -->
+    <ItemGroup Condition="'$(ResolvedCodeAnalysisRuleSet)' != ''">
+      <UpToDateCheckInput Include="$([System.IO.Path]::GetFullPath($(ResolvedCodeAnalysisRuleSet)))" />
```

Comment:
Filed: https://github.com/dotnet/project-system/issues/2521 to reduce the number of design-time builds. 

This change might cause an extra build during startup, can we quickly check to see if there anymore design-time builds with and without this change.?